martypy/ClientMV2.py

```diff
@@ -407,6 +408,99 @@ def is_conn_ready(self) -> bool:
+        for attached_add_on in self.get_add_ons_status().values():
+            if type(attached_add_on) == dict and attached_add_on['name'] == add_on:
+                if attached_add_on['whoAmITypeCode'] in disco_type_codes:
+                    return True
+                else:
+                    raise MartyCommandException(f"The add on name: '{add_on}' is not a valid disco add on. "
+                                                "Please check the add on name in the scratch app -> configure -> add ons")
+        raise MartyCommandException(f"The add on name '{add_on}' is not a valid add on. Please check the add on "
+                                    "name in the scratch app -> configure -> add ons")
+
+    def _valid_hex(self, color_hex: str) -> bool:
+        hex_pattern = re.compile("^([A-Fa-f0-9]{6})$")
+        return bool(hex_pattern.match(color_hex))
+
+    def disco_off(self, add_on: str) -> bool:
+        if self._is_valid_disco_addon(add_on):
+            return self.add_on_query(add_on, bytes.fromhex('01'), 0)
+
+    def disco_pattern(self, pattern: int, add_on: str) -> bool:
+        if pattern == 1:
+                pattern = '10'
+        elif pattern == 2:
+            pattern = '11'
+        else:
+            raise Exception("Pattern must be 1 or 2")
+        if self._is_valid_disco_addon(add_on):
+            return self.add_on_query(add_on, bytes.fromhex(pattern), 0)
+
+    def _region_to_bytes(self, region: Union[str, int]) -> bytes:
+        if region == 'all':
+            region = (2,)
+        else:
+            region = (4, region)
+        return bytes(region)
+
+    def _downscale_color(self, color: Union[tuple, bytes]):
+        return bytes(c//25 for c in color)
+
+    def _parse_color_hex(self, color_hex: str, region: Union[str, int]) -> bytes:
+        input_color = color_hex
+        color_hex = color_hex.lstrip('#')
+        if self._valid_hex(color_hex):
+            color_bytes = bytes.fromhex(color_hex)
+        else:
+            raise MartyCommandException(f"The string '{input_color}' is not a valid hex color code or default color")
+        return color_bytes
+
+    def disco_color(self, color: Union[str, Tuple[int, int, int]], add_on: str, region: Union[int, str]) -> bool:
+        default_colors = {
+            'white'  : 'FFFFFF',
+            'red'    : 'FF0000',
+            'blue'   : '0000FF',
+            'yellow' : 'FFFF00',
+            'green'  : '008000',
+            'teal'   : '008080',
+            'pink'   : 'eb1362',
+            'purple' : '7800c8',
+            'orange' : '961900'
+        }
+        input_color = str(color)
+        if type(color) is str:
+            color = default_colors.get(color.lower(), color)
+            color = self._parse_color_hex(color, region)
+        elif type(color) is tuple:
+            if len(color) != 3:
+                raise MartyCommandException(f'RGB tuple must be 3 numbers, instead of: {input_color}. Please enter a valid color.')
```

Comment:
The original `color` would not have been modified at this stage, would it?
```suggestion
                raise MartyCommandException(f'RGB tuple must be 3 numbers, instead of: {color}. Please enter a valid color.')
```